### PR TITLE
Step: 5 Complete- Added Datastore to store comments; Added functionality to save comments on page

### PR DESCRIPTION
**Pull request by hilakey**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,19 +33,24 @@
...
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
     //Get input from form.
-    String text = getParameter(request, "comments", "");
+    String text = request.getParameter("comments");
+
+    //-----------------------------
+    //Populate list with data.
+    //messages.add(text);
+    //-----------------------------
```
Comment (gagomez) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Save to delete this right?

**Review by gagomez — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (hilakey) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Yes! It was commented out to be deleted so it can be removed.

**Review by sasymer — APPROVED** [review]
All recent changes look good to me!

**Review by jiaxinz-g — COMMENTED** [review]
Sorry about the delay in reviewing, this change looks good to me!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -54,6 +60,25 @@ private String getParameter(HttpServletRequest request, String name, String defa
...
+    List<String> messages = new ArrayList<String>();
+
+    for (Entity entity : results.asIterable()) {
+      String message = (String) entity.getProperty("message");
+        if(comment_null != null){
+         int comment_num = (Integer.valueOf(request.getParameter("limit")));
+          if(messages.size() == comment_num){
+            break;
+          }
+        }
```
Comment (sasymer) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:79]: This block is indented too far I think. It should line up right under the "String message"

Comment (sasymer) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:79]: This block is indented too much (should line up under String message)

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -13,27 +13,19 @@
...
-    greetingsList.appendChild(
-      createListElement(greet[1]));
-    greetingsList.appendChild(
-      createListElement(greet[2]));
-    greetingsList.appendChild(
-      createListElement(greet[3]));
-    greetingsList.appendChild(
-      createListElement(greet[4]));
+    //Build the list of all the comments that have been left on the page.
+    for (let key in greet) {
```
Comment (sasymer) [inline portfolio/src/main/webapp/script.js:23]: Just a suggestion - maybe greet could be named something more representative of being a list/array, like greetings or maybe comments?

File: portfolio/src/main/webapp/style.css [hunk 4]
```diff
@@ -82,18 +93,24 @@ a:visited{
...
+  margin-top: 200px;
   display: block;
   margin-left: auto;
   margin-right: auto;
 }
 
+textarea{
+    font-size: 20px;
+    width: 200px;
+}
```
Comment (sasymer) [inline portfolio/src/main/webapp/style.css:105]: Indented too far

**Pull request merged**